modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sql/GridSqlStatement.java

```diff
@@ -67,4 +73,19 @@ public void limit(GridSqlAst limit) {
     public GridSqlAst limit() {
         return limit;
     }
+
+    /**
+     * @return Delimeter to use.
+     */
+    public static char delimeter() {
```

Comment:
Looks like the delimiter is changed only to check the query in the test and not related somehow to secure output.

Reply:
If we will use '\n' delimiter then queries in system views will look ugly.
I don't think we want to show to user multiline output in `SYS.SQL_QUERIES_HISTORY`.

Agree?

Reply:
But multiline output still will be shown if INCLUDE_SENSITIVE == true

Reply:
No. 

If `INCLUDE_SENSITIVE == true` then we just use query text that came from user. Please, take a look into IgniteH2Indexing line 715
```
        Long qryId = runningQryMgr.register(
            QueryUtils.INCLUDE_SENSITIVE ? qry : sqlWithoutConst(dml.statement()),
            GridCacheQueryType.SQL_FIELDS,
            schemaName,
            true,
            null
        );
```

Reply:
I know it, but text from the user also can contain line delimiters. If we don't want to show to user multiline output we should filter it for all cases on another level.

Reply:
We have 2 cases here:

1. `INCLUDE_SENSITIVE = true` so Ignite just show to the user hist own query text. All OK, nothing to do here.
2. `INCLUDE_SENSITIVE=false` and we should select a format to show query to the user.

We can go with the default formatting, e.g:
```
SELECT
 __Z0.ID,
 __Z0.NAME,
 __Z0.AGE
FROM
 PUBLIC.TST __Z0 
WHERE 
  __Z0.NAME = ?
```
Or we can go with the one-line query, e.g
`SELECT __Z0.ID, __Z0.NAME, __Z0.AGE FROM PUBLIC.TST __Z0 WHERE __Z0.NAME = ?`

I decided to go with the one-line format which a found more pretty.
What is your suggestion?

Reply:
User own query text can also be multiline. I suggest filtering line delimiters on the view level if we want to prohibit multiline output.

Reply:
What is "view level"?
The result of `IgniteH2Indexing#sqlWithoutConst` sent to the 2 places:
  * Running query manager, and to system view and performance statistics from it.
  * `IgniteEvents` API.
It seems to me, that both of these destinations can be treated like "view level".

What do you think?

Reply:
Let's clarify the issue we want to resolve here:

> User own query text can also be multiline.

If we can output the source query body we just do it.
The issue is "What formatting to use if we can't output source user query?"

My suggestion is to use one-line formatting.
What is your suggestion?